packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -323,32 +323,45 @@ module Floats = {
       let continuous: array<float> = []
       let discrete = FloatFloatMap.empty()
 
-      let addData = (count: int, value: float): unit => {
-        if count >= minDiscreteWeight {
-          FloatFloatMap.add(value, count->Belt.Int.toFloat, discrete)
+      // In a run of exactly minDiscreteWeight, the first and last
+      // element indices differ by minDistance.
+      let minDistance = minDiscreteWeight - 1
+
+      let len = length(sortedArray)
```

Comment:
This is a fairly long function that took me a while to wrap my head around. I'd probably be conservative and have things spelled out more. 

Some examples include:
- `len` -> `sortedArrayLen`
- `i` -> `sortedArrayIndex` (or `sortedArrayI`)
- `minDistance` -> `minDistanceOfSameValue`
- `value` -> `indexValue`

Reply:
I can change it, but I find that style harder to read, as `sortedArrayLen` and `sortedArrayIndex` aren't as obviously different as `i` and `len`. There's only one array. Is it really that hard to guess what `i` indexes into?

Reply:
There are several indexes. (i, i0, base, j, lo, mid). This, at very least, seems to have pretty terse naming to me.

I haven't seen much use of `base`, `lo` before. It seems likely that some of this follows lower level programming conventions I'm not used to.

Reply:
Yes, `lo`, `mid`, `hi` is very common for binary searches (using `i` for `hi` partly because declaring mutable variables in rescript is so annoying). I use `i0` here to mean a saved copy of `i`; maybe `iOrig` for `i0` and `iNext` for `j` would be better. `base` is the base of the binary search.